circuit_maintenance_parser/parsers/cogent.py

```diff
@@ -0,0 +1,110 @@
+                        data["account"] = match.group(1)
+                elif line.startswith("Start time:"):
+                    match = re.search(r"Start time: (.*) \([A-Za-z\s]+\) (\d+/\d+/\d+)", line)
+                    if match:
+                        start_str = " ".join(match.groups())
+                elif line.startswith("End time:"):
+                    match = re.search(r"End time: (.*) \([A-Za-z\s]+\) (\d+/\d+/\d+)", line)
+                    if match:
+                        end_str = " ".join(match.groups())
+                elif line.startswith("Cogent customers receiving service"):
+                    match = re.search(r"[^Cogent].*?((\b[A-Z][a-z\s-]+)+, ([A-Za-z-]+[\s-]))", line)
+                    if match:
+                        parsed_timezone = city_timezone(match.group(1).strip())
+                        local_timezone = timezone(parsed_timezone)
+                        # set start time using the local city timezone
+                        start = datetime.strptime(start_str, "%I:%M %p %d/%m/%Y")
+                        local_time = local_timezone.localize(start)
+                        # set start time to UTC
+                        utc_start = local_time.astimezone(UTC)
+                        data["start"] = self.dt2ts(utc_start)
+                        logger.info(
+                            "Mapped start time %s at %s (%s), to %s (UTC)",
+                            start_str,
+                            match.group(1).strip(),
+                            local_timezone,
+                            utc_start,
+                        )
+                        # set end time using the local city timezone
+                        end = datetime.strptime(end_str, "%I:%M %p %d/%m/%Y")
+                        local_time = local_timezone.localize(end)
+                        # set end time to UTC
+                        utc_end = local_time.astimezone(UTC)
+                        data["end"] = self.dt2ts(utc_end)
+                        logger.info(
+                            "Mapped end time %s at %s (%s), to %s (UTC)",
+                            end_str,
+                            match.group(1).strip(),
+                            local_timezone,
+                            utc_end,
+                        )
+                elif line.startswith("Work order number:"):
+                    match = re.search("Work order number: (.*)", line)
+                    if match:
+                        data["maintenance_id"] = match.group(1)
+                elif line.startswith("Order ID(s) impacted:"):
+                    data["circuits"] = []
+                    match = re.search(r"Order ID\(s\) impacted: (.*)", line)
+                    if match:
+                        for circuit_id in match.group(1).split(","):
+                            data["circuits"].append(
+                                CircuitImpact(impact=Impact("OUTAGE"), circuit_id=circuit_id.lstrip())
```

Comment:
```suggestion
                                CircuitImpact(impact=Impact("OUTAGE"), circuit_id=circuit_id.strip())
```
I believe this would cover your more parsing corner cases

Reply:
updated